ikarus/finiteelements/mechanics/materials/hyperelastic/deviatoric/arrudaboyce.hh

```diff
@@ -0,0 +1,198 @@
+// SPDX-FileCopyrightText: 2021-2025 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file arrudaboyce.hh
+ * \brief Implementation of the ArrudaBoyce material model.
+ * \ingroup materials
+ */
+
+#pragma once
+
+#include <ikarus/finiteelements/mechanics/materials/hyperelastic/deviatoric/deviatoricinvariants.hh>
```

Comment:
Maybe we can also create the new material library  within Ikarus for now. Just with cmake but I will have to think that through.

Reply:
Do you mean to have like `#include ikarus/materials/mechanics/...` instead of `#include <ikarus/finiteelements/mechanics/materials/...`? If so, if I recollect correctly, I think our idea was that the material library, at the moment, is only applicable to solve problems in solid mechanics and similar and hence is placed under `mechanics`, because we had the notion that if at all someone implements a material model corresponding to solve thermo-mechanical or `electro-mechanical` problems, then those can be separate folders. Nevertheless, if any such refactoring is needed, it should be in a separate PR and not be mixed with this one. Can you consider this point when looking over #358 and I resolve it now?